src/core/api.c

```diff
@@ -944,6 +944,11 @@ MsQuicStreamSend(
     SendRequest->TotalLength = TotalLength;
     SendRequest->ClientContext = ClientSendContext;
 
+    uint32_t TotalBytesSent = 0;
+    for (uint32_t i = 0; i < BufferCount; ++i) {
+        TotalBytesSent += Buffers[i].Length;
+    }
+
```

Comment:
Put this stuff in QuicStreamSendFlush instead. BTW, `TotalLength` represents what you want already.